src/MessageHandler/EventHandler.php

```diff
@@ -28,7 +29,6 @@ public function __invoke(EventMessage $message): void
             throw new UnrecoverableMessageHandlingException($e->getMessage());
         }
 
-        // @todo: create next message
-        throw new UnrecoverableMessageHandlingException('Not implemented yet');
+        $this->messageBus->dispatch(new ImageMessage($entity->getId(), $entity->getImage()?->getId()));
```

Comment:
Consider moving this inside the `try` block. It may make the code a little easier to understand (and maybe help the code analyzer as well).

Reply:
Not sure I understand this one? What should we try-catch on?

Reply:
I think I would write it like this:

```php
        try {
            $entity = $this->eventFactory->createOrUpdate($item);
            $this->messageBus->dispatch(new ImageMessage($entity->getId(), $entity->getImage()?->getId()));
        } catch (\Exception $e) {
            // @todo: better message.
            throw new UnrecoverableMessageHandlingException($e->getMessage());
        }
```